new-log-viewer/src/components/PopUps/PopUpMessageBox.tsx

```diff
@@ -0,0 +1,120 @@
+import {
+    useContext,
+    useEffect,
+    useState,
+} from "react";
+
+import {
+    Alert,
+    Box,
+    CircularProgress,
+    IconButton,
+    Typography,
+} from "@mui/joy";
+
+import CloseIcon from "@mui/icons-material/Close";
+
+import {
+    NotificationContext,
+    PopUpMessage,
+} from "../../contexts/NotificationContextProvider";
+import {WithId} from "../../typings/common";
+import {LOG_LEVEL} from "../../typings/logs";
+import {DO_NOT_TIMEOUT_VALUE} from "../../typings/notifications";
+
+
+const AUTO_DISMISS_PERCENT_UPDATE_INTERVAL_MILLIS = 50;
+
+interface PopUpMessageProps {
+    message: WithId<PopUpMessage>,
+}
+
+/**
+ * Display a pop-up message in an alert box.
+ *
+ * @param props
+ * @param props.message
+ * @return
+ */
+const PopUpMessageBox = ({message}: PopUpMessageProps) => {
+    const {id, level, message: messageStr, title, timeoutMillis} = message;
+
+    const {handlePopUpMessageClose} = useContext(NotificationContext);
+    const [intervalCount, setIntervalCount] = useState<number>(0);
+
+    const handleCloseButtonClick = () => {
+        handlePopUpMessageClose(id);
+    };
+
+    useEffect(() => {
+        if (DO_NOT_TIMEOUT_VALUE === timeoutMillis) {
+            return () => {};
+        }
+        const intervalId = setInterval(() => {
+            setIntervalCount((c) => c + 1);
+        }, AUTO_DISMISS_PERCENT_UPDATE_INTERVAL_MILLIS);
+
+        return () => {
+            clearInterval(intervalId);
+        };
+    }, [
+        timeoutMillis,
+        handlePopUpMessageClose,
+    ]);
+
+    const color = level >= LOG_LEVEL.ERROR ?
+        "danger" :
+        "primary";
+
+    let percentRemaining = 100;
+    if (DO_NOT_TIMEOUT_VALUE !== timeoutMillis) {
+        const totalIntervals = timeoutMillis / AUTO_DISMISS_PERCENT_UPDATE_INTERVAL_MILLIS;
+        percentRemaining = 100 - (100 * (intervalCount / totalIntervals));
+        if (0 >= percentRemaining) {
+            setTimeout(() => {
+                handlePopUpMessageClose(id);
+            }, 0);
```

Comment:
We want to avoid updating some component while rendering another component. (See the console error if that is removed. )

A more proper way is to move this handlePopUpMessageClose() call inside the useEffect. Let me make the change we can discuss further.